scripts.js

```diff
@@ -113,77 +111,53 @@ document.addEventListener("keyup", () => {
 // check all present asteroids for collision --> may need to convert into function to keep calling
 //increase number of asteroids through increasing asteroid.push
 
-setInterval(() => {
-  const x = 400;
-  // = Math.random() * canvas.width;
+const asteroidCreation = () => {
+  const x = Math.random() * canvas.width;
   asteroids.push(new enemy(x, -10, 30));
-  console.log(asteroids[1]);
-  // for (items of asteroids) {
-  //   return asteroids[items];
-  // }
-}, 2500);
+}
 
-//**STUCK HERE *//
-//asteroid will come down. then check each asteroids x/y etc
-//once asteroid hits ground, remove it
-//repeat step 1
+setInterval(asteroidCreation, 2500);
 
-setInterval(() => {
-  //   for (let i = 0; i < asteroids.length; i++) {
-
-  //     if asteroids[i].y > 500{}
-  //   }
-
-  // if asteroids height is more than 500, remove it from array
-  // if asteroids height is less than 500, keep it in array
-  //do it by creating new array or use filter. filter the easiest. accessing key value of the objects
-  //   if items.y > 500;
-  //   asteroids.splice(0, 1);
-  // }
-
-  //**COLLISION DETECTION NOT WORKING TOO. SOMETHING TO DO WITH THE ARRAY */
-  if (
-    asteroids[0].x < dinosaur[0].x + dinosaur[0].width && //[1] = x, [2]=y, [3]=width, [4] = height
-    asteroids[0].x + asteroids[0].y > dinosaur[0].x &&
-    asteroids[0].y < dinosaur[0].y + dinosaur[0].width &&
-    asteroids[0].y + asteroids[0].width > dinosaur[0].y
-  ) {
-    console.log("true");
+//**STUCK HERE *//
+// asteroid will come down. then check each asteroids x/y etc
+// once asteroid hits ground, remove it
+// repeat step 1
+
+const asteroidRemoval = () => {
+  for (let i = 0; i < asteroids.length; i++) {
+    if (asteroids[i].y > 500) {
+      asteroids.splice(0, 1);
+    }
   }
-}, 100);
-
-//collision detection
-
-// console.log(asteroids);
-
-// {
-//   console.log("true");
-// }
+}
 
-// const checkCollision = () => {
-//   console.log("running");
-//   console.log({ asteroid: asteroids });
-//   console.log({ dinosaur: dinosaur });
-//   if (
-//     asteroids.x < dinosaur.x + dinosaur.width &&
-//     asteroids.x + asteroids.width > dinosaur.x
-//     // &&
-//     // asteroids.y < dinosaur.y + dinosaur.height &&
-//     // asteroids.y + asteroids.height > dinosaur.y
-//   ) {
-//     console.log("true");
-//   }
-// };
+// collision detection
+const checkCollision = () => {
+  if (asteroids.length) {

```

Comment:
You need this check, because when the game starts, there are no asteroids created yet. So the browser will confuse when you ask it to check the asteroid x or y position, because it doesn't exist. Because this fails, it will make the other functions fail too.